virtual_ecosystem/models/animal/functional_group.py

```diff
@@ -129,3 +144,24 @@ def import_functional_groups(
     ]
 
     return functional_group_list
+
+
+def get_functional_group_by_name(
+    functional_groups: tuple[FunctionalGroup, ...], name: str
+) -> FunctionalGroup:
+    """Retrieve a FunctionalGroup by its name from a tuple of FunctionalGroup instances.
+
+    Args:
+        functional_groups: Tuple of FunctionalGroup instances.
+        name: The name of the FunctionalGroup to retrieve.
+
+    Returns:
+        The FunctionalGroup instance with the matching name.
+
+    Raises:
+        ValueError: If no FunctionalGroup with the given name is found.
+    """
+    for fg in functional_groups:
+        if fg.name == name:
+            return fg
+    raise ValueError(f"No FunctionalGroup with name '{name}' found.")
```

Comment:
An alternative:

```suggestion
    return next(fg for fg in functional_groups if fg.name == name)
```

You'll get a `StopIteration` error if there are no matches.

Your way is clear and explicit, so totally up to you which way you do it.

Reply:
Unless the `StopIteration` error is captured and re-raised with some meaningful information, your solution, while concise, might result in very obscure errors.

Reply:
True! I'd only do it this way if I was explicitly checking for errors in the caller.